### ISSUE TITLE
[Feature]: New design for rimusic Part 2: all the tabs and many other things

### ISSUE THREAD

**Issue by jaki444** [fast4x/RiMusic#1812]
### Description

I have created a new design for all the tabs now I will pass you general screenshots with before and afters and zooms.

What the overall view looked like before:
![image_11](https://github.com/fast4x/RiMusic/assets/123010345/c2e99b14-ee4c-44fa-891d-5a13ef4335a7)

What the view looks like with my new design:
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/651d627c-bf0f-45a9-a998-9de1e15ae89f)

Here I will start listing the changes in the library section:
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/c866abf4-5276-4eee-9ca4-bfb38ff56eea)

1. A first change is the removal of favorites,downloads,tops, etc. from the playlist section. (They were only moved not removed after you see).
2. Another change is that now the playlist filter and only the arrow and tendo pressed makes you choose the filter criteria.
3. I decided to remove the pinned section by simply putting the pinned playlists at the top as should be logical.

Here I will start listing the changes in the album section:
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/29b075c1-dea2-4b31-8ed0-514945c369ba)
generally are the same things written before so the summarized filter, the reordered buttons, and the album name position change.

Here I will start listing the changes in the artist section:
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/2fd4a871-f546-4485-8ba2-cf098010aade)
Same changes made to the album section. Placed here for consistency.

Here I will start listing the changes in the song section:

![immagine](https://github.com/fast4x/RiMusic/assets/123010345/fa5d90ed-9f8b-482f-ad26-33082d67f31e)
Here the changes are greater:

1. the action control bar is rebalanced replace the three dots with simply the add to playlist symbol.
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/917e031a-1768-43e8-89af-80a5b9b60c87)
2. It made little sense that the downloaded favorites and top and on the device were on playlists because actually playlists they are not. so I decided to put them in categories of the songs selectable with these buttons:
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/e0d51bbf-30d8-45cb-afb4-e5c73b0ca27e)
Obviously the colored one is intuitively the one selected. 
if you want I create the graphics for each category but with these guidelines I think it's intuitive.

Here I will start listing the changes in the home section:
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/b127215b-6d5d-4c6f-bc7f-7caa5d639be9)
Changes very similar to the other pages. the arrow allows you to quickly change the suggestion from the most listened to to the last most listened to without going through the settings

These are the graphical changes to the control and playback bar:
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/02eee521-0288-462f-b73c-60a8272beea4)
the control bar has been rhessa more rounded and smaller and shaded to give a sense of depth and order and not to create a sharp detachment given the not-too-equal colors.

Another change is in the playback bar, which is now not attached, so it floats. it has a shadow to be deeper and rounded to recreate a more pleasing and modern effect.
![immagine](https://github.com/fast4x/RiMusic/assets/123010345/a9384569-85f2-4f96-bace-ad5bd4f17e32)

Of course, all of this is my own graphic interpretation, which must suit fast4x in the first place and also you. Anyway let me know if you can like it and if I need to make changes or if it sucks for you. thank you.

### Suggested Solution

.

**Comment by jaki444** [comment]
If you want to zommare better I give you the link to visualize. there is both the old and the new design : https://www.figma.com/file/5UxskN9vjA96wpR8T96j6D/RIMUSIC--new?type=design&node-id=0%3A1&mode=design&t=0pbZLXwiumrll4d1-1

**Comment by AquamarineDreams** [comment]
I don't know what to think about this one. I really like the new library, and I don't think that moving everything to the song section is really necessary. Although I agree with the pinned playlist section. However I would keep the pin icon on the playlists, instead of just moving them to the top.

About the general changes of the interface, you moved the page text to the left. You need to keep in mind that, probably, most people use the controll bar on the left (because is the default option).
I didn't like the summarized filter, because seems less intuitive, and would probably bring an extra menu to this app function.

Really liked the floating player.

In short, it's pretty good, but in my opinion, some parts need to be rethought

**Comment by JeePopuwu** [comment]
Apart from Data-tron, you are the one who gives the most innovative ideas, and they are surprising. My respects, I hope Fast likes the ideas, and I would like to see all of this soon.

**Comment by jaki444** [comment]
> I don't know what to think about this one. I really like the new library, and I don't think that moving everything to the song section is really necessary. Although I agree with the pinned playlist section.
> 
> About the general changes of the interface, you moved the page text to the left. You need to keep in mind that, probably, most people use the controll bar on the left (because is the default option). I didn't like the summarized filter, because seems less intuitive, and would probably bring an extra menu to this app function.
> 
> Really liked the floating player.
> 
> In short, it's pretty good, but in my opinion, some parts need to be rethought

Some things you say I agree with others less so we'll see what @fast4x  and others think about it.

**Comment by jaki444** [comment]
> I don't know what to think about this one. I really like the new library, and I don't think that moving everything to the song section is really necessary. Although I agree with the pinned playlist section.
> 
> About the general changes of the interface, you moved the page text to the left. You need to keep in mind that, probably, most people use the controll bar on the left (because is the default option). I didn't like the summarized filter, because seems less intuitive, and would probably bring an extra menu to this app function.
> 
> Really liked the floating player.
> 
> In short, it's pretty good, but in my opinion, some parts need to be rethought

As soon as I have time I'll redo the design with both the bar on the right, below and on the left following some of your guidelines. the ones I leave then I'll explain why as soon as I finish it.

**Comment by jaki444** [comment]
> Apart from Data-tron, you are the one who gives the most innovative ideas, and they are surprising. My respects, I hope Fast likes the ideas, and I would like to see all of this soon.

Really thank you very much.

**Comment by jaki444** [comment]
> Apart from Data-tron, you are the one who gives the most innovative ideas, and they are surprising. My respects, I hope Fast likes the ideas, and I would like to see all of this soon.

If you have any refinements for me please tell me at least when I retouch the design I can put yours in as well.

**Comment by jaki444** [comment]
However, before touching up the design I will wait for today's versions which I think fast4x surprised us in a positive sense.

**Comment by fast4x** [comment]
I wait for the release of the new version and I read it carefully, but at first glance it seems amazing. Meanwhile, I agree with the compliments you have already received. Thanks

**Comment by AquamarineDreams** [comment]
> However, before touching up the design I will wait for today's versions which I think fast4x surprised us in a positive sense.

No worries bro, go on your own time. Thanks for listening to my suggestions btw. The fact that you are spending your time to make this app even better is already enough.

I have great hopes for today's version.

**Comment by fast4x** [comment]
To be honest, I like everything. I didn’t quite understand how you want to display sorting options. For example by clicking on the up arrow, I would change the sorting but the user could not select the sort by title, artist etc...
One solution to evaluate might be to use a long click on the arrow to display sorting options...
Nice work, congratulations

**Comment by jaki444** [comment]
Ah ok thank you. Yes one solution is to hold. Thank you again.

**Comment by jaki444** [comment]
I didn't see and answer before because I was really busy, I don't know if you are already working to put it up anyway I am making a diagram with all the graphics to make it even clearer if you need it.

**Comment by fast4x** [comment]
I haven’t started yet, if you’re happy to preview how it’s going to turn out, it’s not bad. Thanks

**Comment by fast4x** [comment]
Started...

**Comment by aneesh1122** [comment]
> Started...

![friendshappy-grinchmeme](https://github.com/fast4x/RiMusic/assets/45353488/a4777d5a-2c8b-4cb4-9dcc-f3274ca9e4e3)

**Comment by fast4x** [comment]
Work.... in... progress..... 

https://github.com/fast4x/RiMusic/assets/1484476/edc321ff-1bc4-4a89-83cc-50794fb2d3cb

**Comment by fast4x** [comment]
and in ViMusic mode

https://github.com/fast4x/RiMusic/assets/1484476/8b8c4747-eb42-4099-96a5-c834b3b1c197

**Comment by AquamarineDreams** [comment]
> Work.... in... progress.....
> Screen_recording_20240520_093453.mp4

Top 70 is crazy

**Comment by fast4x** [comment]
> > Work.... in... progress.....
> > Screen_recording_20240520_093453.mp4
> 
> Top 70 is crazy

In fact I’m crazy 🤣🤣🤣

**Comment by fast4x** [comment]
> > Work.... in... progress.....
> > 
> >  Screen_recording_20240520_093453.mp4
> 
> - **Songs page**
> 
> Hey fast4x, great job. In the video you've posted some options are missing. I don't know what options are available inside the 3 dots.
> 
> I suggest keeping these on the top of the page
> "Sort, Download all, Remove all, Shuffle, search, locate playing song" (These are what I use the most)
> 
> Rest of them inside the 3 dots:
> "Smart recommendations, Hide/unhide, show on device songs, play next, enqueue, add to playlist, export playlist, select/deselect"
> (These are what I rarely use)
> 
> feel free to change the above list according to you.
> 
> - **Playlists page**
> 
> You don't really need an another section for 'Pinned playlists'. Just make the pinned playlists appear in the list before other playlists like shown in the below
> ![329260886-c866abf4-5276-4eee-9ca4-bfb38ff56eea](https://github.com/fast4x/RiMusic/assets/45353488/126381d0-88db-4a28-aa06-e3d38e5b5200)
> First 3 pinned playlists appear first and then the rest of the unpinned playlists.

ok wok in progress

**Comment by fast4x** [comment]
> Playlists page\nYou don't really need an another section for 'Pinned playlists'. Just make the pinned playlists appear in the list before other playlists like shown in the below

For now isn't simple...

**Comment by aneesh1122** [comment]
> > Work.... in... progress.....
> > Screen_recording_20240520_093453.mp4
> 
> Top 70 is crazy

Me with Top 100
![IMG_20240520_172438.jpg](https://github.com/fast4x/RiMusic/assets/45353488/2632fe88-5377-4b31-8fce-aa1da0cc5a1e)

**Comment by jaki444** [comment]
> and in ViMusic mode
> Screen_recording_20240520_093809.mp4

You are doing a really good job.

**Comment by fast4x** [comment]
Preview... my pleasure share with all work in progress...
Of course will be a couple of bugs... but it's a developer life... :)

https://github.com/fast4x/RiMusic/assets/1484476/338c9d17-2c14-415f-bf6a-62e40cdc89e6

**Comment by fast4x** [comment]
Change quick pics...

https://github.com/fast4x/RiMusic/assets/1484476/b391ba14-a6fe-4d4b-b302-c10673d1d1a4

**Comment by JeePopuwu** [comment]
> Vista previa... es un placer compartirla con todo el trabajo en progreso... Por supuesto, habrá un par de errores... pero es una vida de desarrollador... :)
> 
>  Grabación_de_pantalla_20240521_163954.mp4

WE LOVE YOU FAST

Fast:
![562171502a537f78b37e5ae72e30036f](https://github.com/fast4x/RiMusic/assets/159836335/ac67b509-d545-4bad-8526-c3ab421892d5)

**Comment by fast4x** [comment]
Also i love you too
![giphy (1)](https://github.com/fast4x/RiMusic/assets/1484476/b6084f19-91ef-4b7b-8f6c-81e4c3917fde)

**Comment by aneesh1122** [comment]
> Preview... my pleasure share with all work in progress...
> Of course will be a couple of bugs... but it's a developer life... :)
> 
> https://github.com/fast4x/RiMusic/assets/1484476/338c9d17-2c14-415f-bf6a-62e40cdc89e6
> 
> 

YOU'VE FINALLY DONE IT

You've finally put the mini player above the navigation bar 😅

**Comment by fast4x** [comment]
yyyeeessss

**Comment by AquamarineDreams** [comment]
🔥🔥🔥

**Comment by dAtA-TRoN** [comment]
> YOU'VE FINALLY DONE IT\n\nYou've finally put the mini player above the navigation bar 😅

😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱😱

https://github.com/fast4x/RiMusic/assets/71184627/aae369b8-a95b-4564-8e8f-affd24ca2131

**Comment by fast4x** [comment]
not only that, the mini player can be shown everywhere and... maybe you can move up...

**Comment by dAtA-TRoN** [comment]
> not only that, the mini player can be shown everywhere and... maybe you can move up...

Oh God, this is such incredibly great news. RiMusic always felt kind of broken to me because of this -until now. Finally, everything will be alright 🥳! Thank you Mr fast4x. This is awesome!!!

**Comment by dAtA-TRoN** [comment]
> 🔥🔥🔥

🔥🔥🔥 indeed.

**Comment by dAtA-TRoN** [comment]
> ![Screenshot_2024-05-21-22-45-32-476_com.brave.browser.png](https://github.com/fast4x/RiMusic/assets/45353488/15bcd40f-1d00-4b87-b6b8-971690a2ff05)
> 
> @dAtA-TRoN 

Yeah, isn't it awesome 🤯!?

**Comment by fast4x** [comment]
now... player in settings page...
![Screenshot_2024-05-21-19-17-53-096_it.fast4x.rimusic.jpg](https://github.com/fast4x/RiMusic/assets/1484476/c964750b-4886-4cb5-be2f-75f552414614)

**Comment by aneesh1122** [comment]
> @aneesh1122 
> 
> Yeah, isn't it awesome 🤯!? 

Wait until you see what's next fast4x is cooking

**Comment by ikanakova** [comment]
fast4x:

![fast4x](https://media2.giphy.com/media/l4q8hciiYNT5RGi4w/giphy.gif?cid=6c09b952xlkv8h3a0wlbyq413ywjp3gn7an27fin6q298elg&ep=v1_internal_gif_by_id&rid=giphy.gif&ct=g)

**Comment by aneesh1122** [comment]
@fast4x after this update you should change screenshots in the main page. The app is changed a lot now.

**Comment by AquamarineDreams** [comment]
> @fast4x after this update you should change screenshots in the main page and in your website. The app is changed a lot now.

Fr

**Comment by fast4x** [comment]
> @fast4x after this update you should change screenshots in the main page and in your website. The app is changed a lot now.

yes you are right

**Comment by fast4x** [comment]
> > @fast4x after this update you should change screenshots in the main page and in your website. The app is changed a lot now.
> 
> Fr

yes you are right

**Comment by jaki444** [comment]
fast4x you are the best

**Comment by fast4x** [comment]
Thanks to these changes, I also started to re-engineer the player, preparing the ground for advanced features that can be developed later. These changes require work on RiMusic surgery, so there will be many bugs initially, such as a patient leaving the hospital after a surgical operation...

**Comment by fast4x** [comment]
............... initial ..................

![giphy (2)](https://github.com/fast4x/RiMusic/assets/1484476/18d7fdc7-cc1f-4d80-a1fd-243bdeb39cf9)

............. when complete .............

![giphy (3)](https://github.com/fast4x/RiMusic/assets/1484476/c7edadb5-36c4-45b1-8c82-e51810c5d8b3)

**Comment by AquamarineDreams** [comment]
Debugging must be a nightmare 😅

**Comment by aneesh1122** [comment]
> Debugging must be a nightmare 😅

You're wrong.

Fast4x is the nightmare for debugging 😂

**Comment by AquamarineDreams** [comment]
> > Debugging must be a nightmare 😅
> 
> You're wrong.
> 
> Fast4x is the nightmare for debugging 😂

Bros such a good programer that there's no need for debugging 😅

**Comment by fast4x** [comment]
Implemented, available from version 0.6.36

**Comment by jaki444** [comment]
Fast4x this update is wonderful. Really amazing work.
